core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -4768,6 +4768,112 @@ public String getDefaultScheme() {
     }
   }
 
+  @Test
+  public void testApplyDefaultServiceConfigIfInitialNameResolutionFails() throws Exception {
+    LoadBalancerRegistry.getDefaultRegistry().register(mockLoadBalancerProvider);
+    timer.forwardNanos(1234);
+    channelBuilder.maxTraceEvents(10);
+    try {
+      Status resolutionError = Status.UNAVAILABLE
+              .withDescription("Initial Name Resolution error, using default service config");
+      FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory
+              .Builder(expectedUri).setServers(Collections.singletonList(new
+              EquivalentAddressGroup(socketAddress))).setError(resolutionError).build();
+      channelBuilder.nameResolverFactory(nameResolverFactory);
+      Map<String, Object> defaultServiceConfig =
+              parseConfig("{\"methodConfig\":[{"
+                      + "\"name\":[{\"service\":\"SimpleService1\"}],"
+                      + "\"waitForReady\":true}]}");
+      channelBuilder.defaultServiceConfig(defaultServiceConfig);
+      ManagedChannelServiceConfig managedChannelServiceConfig =
+          createManagedChannelServiceConfig(defaultServiceConfig, null);
+      nameResolverFactory.nextConfigOrError.set(
```

Comment:
What is the purpose of setting nextConfigOrError? I don't see it being used in the course of the test.

Reply:
This will give the info like we are getting the defaultServiceConfig details which are internally setting to through ManagedChannelServiceConfig and setting this into nameResolverFactory which we are using below.

defaultServiceConfig -> ManagedChannelServiceConfig -> nameResolverFactory.nextConfigOrError.

do you thinks its not required? because I have seen some existing references similar to this in existing UT's in the same class.

Reply:
If existing UTs are setting it they are using it. I don't see how your test uses it.

Reply:
Fixed and Removed this setup/config step as it's not harming test flow.